### DISCUSSION TITLE
Abstract Types not Working too well (Class). wrong / not allowed implementation?

### DISCUSSION THREAD

**Discussion by MCole-K9** [sveltejs/kit#6136] [Q&A] [answered]
So i'm new to typescript and Svelte(kit). I've been experimenting with Classes and Interfaces.  I've been mainly refactoring some bits and pieces to of the SvelteKit Demo App to use class and interfaces instead of Just Types to Accomplish some OOP stuff, like having methods to conveniently transform / get / set data in an object. Here's what i've been doing.

 
![image](https://user-images.githubusercontent.com/81397122/185775209-771ab7b9-69ab-491f-a3cd-818c6e0645ac.png)

![image](https://user-images.githubusercontent.com/81397122/185775278-3e935c63-e508-405e-b23b-625922e102ca.png)

Transforming the data.todos to Todo instances in order to  use the object methods

![image](https://user-images.githubusercontent.com/81397122/185775309-802417a6-a580-4f31-b571-10574ee3130e.png)

![image](https://user-images.githubusercontent.com/81397122/185775472-5a334242-5faf-4eb6-9722-a156a16cbd7c.png)

It works. But not smoothly. I have to hit refresh on the browser to see changes reflected..

So my question is, Is this implementation valid?

**Comment by david-plugge** [comment, marked as answer]
I think the only missing Part is to make
`const todos = data.todos.map...` a reactive statement:
`$: todos = data.todos.map...`

**Reply by MCole-K9** [reply]
Thanks much.. This makes absolute sense.
